config/options.go

```diff
@@ -64,8 +66,18 @@ func Reporter(reporter jaeger.Reporter) Option {
 
 // Observer can be registered with the Tracer to receive notifications about new Spans.
 func Observer(observer jaeger.Observer) Option {
+	old := jaeger.OldObserver{Obs: observer}
```

Comment:
it's not good to make OldObserver public. There are two ways we can handle this:
  1. duplicate ContribObserver option on the Tracer and do the wrapping there and only there, thus making oldObserver private
  1. or define an `internal/observer` package and declare a public type there - by convention people should not be depending on APIs under `internal`